api/src/opentrons/protocol_engine/state/commands.py

```diff
@@ -30,6 +32,8 @@ class CommandState:
     """State of all protocol engine command resources."""
 
     is_running: bool
+    is_gracefully_finished: bool
+    is_hardware_stopped: bool
     stop_requested: bool
```

Comment:
Separately from the `is_running` naming question, here's a shot at describing what each of these means now. Does this look right?

```suggestion
    # Whether the engine is consuming commands out of the queue.
    # Note that this can be False while the hardware is still in motion.
    is_running: bool

    # Whether the thing feeding commands to the Engine is done doing that,
    # because it successfully reached the end of the protocol. 
    # Once set to True, never set back to False.
    is_gracefully_finished: bool

    # Whether the hardware has ceased moving for the lifetime of this Engine.
    # Once set to True, never set back to False.
    is_hardware_stopped: bool

    # Whether something has requested that the hardware cease moving ASAP,
    # and stay ceased for the lifetime of this Engine.
    # Once set to True, never set back to False.
    stop_requested: bool

```